File: javascript/1752. Check if Array Is Sorted and Rotated.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var check = function(nums) {
    var decrease = 0;
    var len = nums.length;
    if (nums[0] < nums[len - 1])
        ++decrease;
    
    for (var i = 0; i < len - 1; ++i) {
        if (nums[i] > nums[i + 1]) {
            if (++decrease == 2)
               return false;
        }
    }
    return true;
};

File: java/1752. Check if Array Is Sorted and Rotated.java
class Solution {
    public boolean check(int[] nums) {
        int ans = 0;
        int size = nums.length;
        if (nums[0] < nums[size - 1])
           ++ans;
        for (int i = 1; i < size; ++i) {
            if (nums[i] < nums[i - 1]) {
                if (++ans > 1)
                   return false;
            }
        }
        return true;
    }
}